### ISSUE TITLE
Validate_Json() and parse_json does not catch missing start "{" squirly bracket

### ISSUE THREAD

**Issue by LarsGut** [godotengine/godot#40794]
**Godot version:**

3.2.2.stable.official

**OS/device including version:**

Windows 10

**Issue description:**

When testing Validate_Json() and parse_json, i found that it does not catch the missing start '{' ,but gladly continues and thinks its a valid JSON string, althoug prints out some thing, but not a JSON formated string.

**Steps to reproduce:**
1 remove the first squirly bracket
2 run example, no exception is thrown

**Minimal reproduction project:**

```
# correct string '{ "apa":12345,"epa":12345 }'

onready var parseTest ={}
onready var jsonString = ' "apa":12345,"epa":12345 }
func _ready() -> void:
	
	var a = validate_json(jsonString)
	
	if not a:
		parseTest = parse_json(jsonString)
		print("---------------------")
		print(parseTest)
	else:
		print(a)
```

**Comment by Xrayez** [comment]
I confirm this on the core level in #40795. 🙂

**Comment by akien-mga** [comment]
Fixed by #40804.
